pandas/core/series.py

```diff
@@ -2177,6 +2177,49 @@ def map_f(values, f):
         return self._constructor(new_values,
                                  index=self.index).__finalize__(self)
 
+    def _gotitem(self, key, ndim, subset=None):
+        """
+        sub-classes to define
+        return a sliced object
+
+        Parameters
+        ----------
+        key : string / list of selections
+        ndim : 1,2
+            requested ndim of result
+        subset : object, default None
+            subset to act on
+        """
+        return self
+
+    @Appender(generic._shared_docs['aggregate'] % _shared_doc_kwargs)
+    def aggregate(self, func, axis=0, *args, **kwargs):
+        axis = self._get_axis_number(axis)
+        result, how = self._aggregate(func, *args, **kwargs)
+        if result is None:
+
+            # we can be called from an inner function which
+            # passes this meta-data
+            kwargs.pop('_axis', None)
+            kwargs.pop('_level', None)
+
+            # try a regular apply, this evaluates lambdas
+            # row-by-row; however if the lambda is expected a Series
+            # expression, e.g.: lambda x: x-x.quantile(0.25)
+            # this will fail, so we can try a vectorized evaluation
+
+            # we cannot FIRST try the vectorized evaluation, becuase
+            # then .agg and .apply would have different semantics if the
+            # operation is actually defined on the Series, e.g. str
+            try:
+                result = self.apply(func, *args, **kwargs)
+            except (ValueError, AttributeError, TypeError):
+                result = func(self, *args, **kwargs)
```

Comment:
yes, something like this (its because Series will row-by-row application)

```
In [4]: Series(range(5)).apply(lambda x: x-x.min())
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-e65bceda88a4> in <module>()
----> 1 Series(range(5)).apply(lambda x: x-x.min())

/Users/jreback/miniconda3/envs/pandas/lib/python3.5/site-packages/pandas/core/series.py in apply(self, func, convert_dtype, args, **kwds)
   2290             else:
   2291                 values = self.asobject
-> 2292                 mapped = lib.map_infer(values, f, convert=convert_dtype)
   2293 
   2294         if len(mapped) and isinstance(mapped[0], Series):

pandas/src/inference.pyx in pandas.lib.map_infer (pandas/lib.c:66116)()

<ipython-input-4-e65bceda88a4> in <lambda>(x)
----> 1 Series(range(5)).apply(lambda x: x-x.min())

AttributeError: 'int' object has no attribute 'min'
```